server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -104,10 +104,16 @@
     @LazyCollection(LazyCollectionOption.EXTRA) // allows .size() without loading all data
     private List<ProductContent> productContent;
 
+    /*
+     * hibernate persists empty set as null, and tries to fetch
+     * dependentProductIds upon a fetch when we lazy load. to fix this, we eager
+     * fetch.
+     */
     @ElementCollection
     @CollectionTable(name = "cpo_product_dependent_products",
                      joinColumns = @JoinColumn(name = "product_uuid"))
     @Column(name = "element")
+    @LazyCollection(LazyCollectionOption.FALSE)
```

Comment:
per my comment above I don't think it is that were setting this as null, but hibernate is persisting it as null.

I think we are hitting: https://forum.hibernate.org/viewtopic.php?f=1&t=1011750
( but for a Set and not a list. )
Ive tried and contemplated various approaches ( detaching object from
hibernate, creating a new session in the getSubscription call, etc)
with no success.


using @OrderColumn doesnt work either cause we have a set of strings.